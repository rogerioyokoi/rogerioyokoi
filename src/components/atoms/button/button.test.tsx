import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { axe } from 'vitest-axe';
import Button, { ButtonProps } from './button';

describe('Components > Atoms > Button', () => {
  const renderButton = (props: ButtonProps) => {
    return render(<Button {...props}>My Button Test</Button>);
  };

  describe('Snapshot', () => {
    it('Should match snapshot basic button', () => {
      const { container } = renderButton({});

      expect(container).toMatchSnapshot();
    });

    it('Should match snapshot basic anchor', () => {
      const { container } = renderButton({ href: 'https://example.com' });

      expect(container).toMatchSnapshot();
    });
  });

  describe('Accessibility Tests (A11y)', () => {
    it('Should not have accessibility violations on basic button', async () => {
      const { container } = renderButton({});

      expect(await axe(container)).toHaveNoViolations();
    });

    it('Should not have accessibility violations on basic anchor', async () => {
      const { container } = renderButton({ href: 'https://example.com' });

      expect(await axe(container)).toHaveNoViolations();
    });
  });

  describe('Rendering', () => {
    it('Should render anchor correctly', () => {
      const { getByTestId, queryByTestId } = renderButton({ href: 'https://example.com' });

      const iElmAnchor = getByTestId('button-anchor');
      const iElmButton = queryByTestId('button-type');

      expect(iElmAnchor).toBeInTheDocument();
      expect(iElmButton).toBeNull();
    });

    it('Should render button correctly', () => {
      const { getByTestId, queryByTestId } = renderButton({});

      const iElmAnchor = queryByTestId('button-anchor');
      const iElmButton = getByTestId('button-type');

      expect(iElmAnchor).toBeNull();
      expect(iElmButton).toBeInTheDocument();
    });
  });

  // describe('CSS Classes', () => {
  //   it('Should apply correct classes to title', () => {
  //     const { getByTestId } = render(<ComponentName title={mockTitleProp} />);
  //     const headingElement = getByTestId(iElmTitleTestId);

  //     expect(headingElement).toHaveClass('text-amber-500 text-3xl lg:text-4xl');
  //   });
  // });
});
