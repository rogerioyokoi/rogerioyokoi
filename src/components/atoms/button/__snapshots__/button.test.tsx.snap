// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components > Atoms > Button > Snapshot > Should match snapshot basic anchor 1`] = `
<div>
  <a
    class="flex items-center justify-between gap-4 rounded-full transition-colors ease-in-out duration-300 font-medium pl-6 pr-6 text-lg bg-amber-400 hover:bg-amber-600 text-slate-900"
    data-testid="button-anchor"
    href="https://example.com"
  >
     
    <span
      class="w-100"
    >
      My Button Test
    </span>
     
  </a>
</div>
`;

exports[`Components > Atoms > Button > Snapshot > Should match snapshot basic button 1`] = `
<div>
  <button
    class="flex items-center justify-between gap-4 rounded-full transition-colors ease-in-out duration-300 font-medium pl-6 pr-6 text-lg bg-amber-400 hover:bg-amber-600 text-slate-900"
    data-testid="button-type"
  >
     
    <span
      class="w-100"
    >
      My Button Test
    </span>
     
  </button>
</div>
`;
